appserver/persistence/common/src/main/java/org/glassfish/persistence/common/PersistenceHelper.java

```diff
@@ -44,10 +45,25 @@ public static DataSource lookupPMResource(ConnectorRuntime connectorRuntime, Dep
     }
 
     private static ResourceInfo getResourceInfo(DeploymentContext ctx, SimpleJndiName dataSourceName) {
+        dataSourceName = translateResourceReference(ctx, dataSourceName);
         if (dataSourceName.isJavaApp()) {
             String applicationName = ctx.getCommandParameters(OpsParams.class).name();
             return new ResourceInfo(dataSourceName, applicationName);
         }
         return new ResourceInfo(dataSourceName);
     }
+
+    private static SimpleJndiName translateResourceReference(DeploymentContext ctx, SimpleJndiName dataSourceName) {
+        final BundleDescriptor currentBundle = DOLUtils.getCurrentBundleForContext(ctx);
+        if (currentBundle instanceof ResourceReferenceContainer) {
+            ResourceReferenceContainer referenceContainer = (ResourceReferenceContainer)currentBundle;
+            try {
+                return referenceContainer.getResourceReferenceByName(dataSourceName.toString()).getJndiName();
+            } catch (IllegalArgumentException e) {
+                return dataSourceName;
```

Comment:
It would be good to log the exception as a warning at least.

Reply:
Not here. This is a valid condition - if the datasource name is a proper JNDI and not a reference, just use it without any change. It would only pollute logs if it was a warning. I can add a debug logging.

The `getResourceReferenceByName` gives an exception if resource not found. Ideally, I would rather use `findResourceReferenceByName`, which returns a `null`, but that one is `protected` and only in `WebBundleDescriptor`, so I rather used a method on the interface `ResourceReferenceContainer`.

Reply:
Ok, debug/trace is good then. Sometimes I am hunting things around DOL and XML/annotations and any such log is useful.